controllers/clusterclassutil/indexer.go

```diff
@@ -0,0 +1,51 @@
+/*
```

Comment:
Usually `util` packages end up being a "catch-all", have we thought about a different name?

Reply:
Do we need a separate package? Afaik usually we just inlined those funcs in the past

Reply:
Hm, maybe `clusterclassindex`? I wonder if we need a separate package.

Reply:
What about keeping those two func in the same file of the ClusterTopology controller given that it is the only consumer?

Reply:
Suggestion on a tangential topic, lets stuff the util functions in a `topology.go` file. It can house the name hashing function that Fabrizio was talking about. Also, maybe adding the topology label thingy can be done in there too.

Reply:
We've been creating a lot of indexes lately around and they're all spread around in packages, should we have an `indexes/` package under the API types instead? 

cc @CecileRobertMichon

Reply:
Ref: https://github.com/kubernetes-sigs/cluster-api/pull/5004